plugins/out_forward/forward.c

```diff
@@ -647,25 +645,16 @@ static int forward_config_ha(const char *upstream_file,
         else if (tmp) {
             fc->shared_key = flb_sds_create(tmp);
         }
-        else {
-            fc->shared_key = NULL;
-        }
 
         tmp = flb_upstream_node_get_property("username", node);
         if (tmp) {
             fc->username = tmp;
         }
-        else {
-            fc->username = "";
```

Comment:
This assignment should not removed; This will break deployments that
assume `username` defaults to empty string.

(In particular, this is because of ha-proxy setting is not being subject to flb_output_config_map_set() in the current implementation)